stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/plugin/logic/channel/internal/ChannelLogic.kt

```diff
@@ -399,34 +399,34 @@ internal class ChannelLogic(
      *
      * @param message [Message].
      */
-    internal fun incrementUnreadCountIfNecessary(message: Message) {
+    private fun incrementUnreadCountIfNecessary(message: Message) {
         val currentUserId = globalMutableState.user.value?.id ?: return
+        val unreadCount: Int = mutableState._unreadCount.value
```

Comment:
Here I'm dealing with race conditions with immutability instead of locks. I don't pick the value of unreadCount **after** the `shouldIncrementUnreadCount` evaluation. So even if the change in the `unreadCount` happens twice, if will happen for the same value, because the snapshot of the data already happened.